src/transport/Dat.ts

```diff
@@ -1,92 +1,254 @@
+//                     if (err) throw err
+//                     logs.push('Done')
+//                     callback(logs)
+//                     return envelope
+//                 })
+//                 progress.on('put', function (src) {
+//                     logs.push('Downloading', src.name)
+//                 })
+//             }
+//         })
+
+//     }
+
+    // createOutgoingArchive(id:string, archivePath:string): Promise<DatNode> {
+    //     return this.createDatNode(id, archivePath, {watch:true})
+    // }
+
+    // createTransientOutgoingArchive(id:string, archivePath:string): Promise<DatNode> {
+    //     return this.createDatNode(id, archivePath, {watch:true, temp:true})
+    // }
+
+    // createIncomingArchive(id:string, key:string, archivePath:string): Promise<DatNode> {
+    //     return this.createDatNode(id, archivePath, { key: key})
+    // }
+
+    // createTransientIncomingArchive(id:string, key:string): Promise<DatNode> {
+    //     return this.createDatNode(id, ram, { key: key, temp: true })
+    // }
+
+    public getDatNode(id:string) {
+        let me = this
+        return me._datnodes[id]
+    }
 
-            var progress = dat.importFiles(src, {
-                ignore: ignores || []
-            }, function (err) {
-                if (err) throw err
-                logs.push("Done importing")
-                logs.push("Archive size: " + dat.archive.content.byteLength)
-                return callback(dat.key.toString('hex'), logs)
+    public createDatNode(id:string, archive:string|{}, options:{}): Promise<DatNode> {
+        let me = this
+        return new Promise<DatNode>(function(resolve, reject) {
+            if (me._datnodes[id]) {
+                return reject(new Error("DatNode with ID " + id + " already exists"))
+            } else {
+                me._datnodes[id] = new DatNode(id, archive, options)
+                return resolve(me._datnodes[id])
+            }
+        })
+    }
+
+    public disposeDatNode(id:string): Promise<DatNode> {
+        let me = this
+        return new Promise<DatNode>(function(resolve, reject) {
+            let datnode:DatNode = me.getDatNode(id);
+            delete me._datnodes[id];
+            if (datnode) {
+                try {
+                    datnode.close()
+                } catch (error) {
+                    console.error("Error while closing '" + id)
+                    // console.error(error)
+                    return reject(datnode)
+                }
+            }
+            return resolve(datnode)
+        })
+    }
+
+    public shutdown(): Promise<any> {
+        let me = this
+        var subPromises = []
+        Object.keys(me._datnodes).forEach(function (id) {
+            subPromises.push(me.disposeDatNode(id)
+                .catch(function(error) {
+                    console.error("Unexpected error while closing datnode " + id )
+                    // console.error(error)
```

Comment:
Request: Please remove this commented out line.

Reply:
Fuck. You got me. Need that linter...

Reply:
👍